splunktaucclib/data_collection/ta_data_client.py

```diff
@@ -32,9 +32,8 @@ def build_event(
 ):
     if is_unbroken is False and is_done is True:
         raise Exception("is_unbroken=False is_done=True is invalid")
-    return tdc.event_tuple._make(
-        [host, source, sourcetype, time, index, raw_data, is_unbroken, is_done]
-    )
+    writer = ew.ClassicEventWriter()
```

Comment:
Creating an instance of writers every time when we're building events could not be effective. We could:
- change this method to class method and store event writer as a singleton
- change this method in `solnlib.ClassicEventWriter` to class method (or even better - add new class method  and mark current as deprecated)
- leave it as it is.

Any opinions?

Reply:
+1 for singleton option.

Second option is the best but might take longer, but think changing `ClassicEventWriter.create_event` to `@classmethod` should be enough as it doesn't change the behaviour for existing dependencies - the method is immutable and class method just widens the access from 
`obj().method() `
to 
`obj().method(), obj.method()`

Reply:
I used yet another option with closure, I think it will be ok.